internal/allocator/allocator_test.go

```diff
@@ -1226,6 +1228,131 @@ func TestPoolCount(t *testing.T) {
 	}
 }
 
+func TestPoolMetrics(t *testing.T) {
+	alloc := New()
+	if err := alloc.SetPools(map[string]*config.Pool{
+		"test": {
+			AutoAssign: true,
+			CIDR: []*net.IPNet{
+				ipnet("1.2.3.4/30"),
+				ipnet("1000::4/126"),
+			},
+		},
+	}); err != nil {
+		t.Fatalf("SetPools: %s", err)
+	}
+
+	tests := []struct {
+		desc       string
+		svc        string
+		ip         string
+		ports      []Port
+		sharingKey string
+		backendKey string
+		ipsInUse   int
```

Comment:
Can you use float64 here, so no type conversion is [needed later][link]?

[link]: https://github.com/metallb/metallb/pull/627/files#diff-4febd49fd5842f0b4e763b000130e8b7R1323